lib/linalg/Matrix.cpp

```diff
@@ -1832,6 +1832,64 @@ Matrix::orthogonalize()
     }
 }
 
+void 
+Matrix::rescale_rows_max()
+{
+	// Rescale every matrix row by its maximum absolute value.
+	for (int i = 0; i < d_num_rows; i++)
+	{
+		// Find the row's max absolute value.
+		double local_max = fabs(item(i, 0));
+		for (int j = 1; j < d_num_cols; j++)
+		{
+			if (fabs(item(i, j)) > local_max)
+				local_max = fabs(item(i, j));
+		}
+
+		// Get the max of all processes, if applicable.
+		double global_max = local_max;
+		if (d_num_procs > 1)
+			MPI_Allreduce(&local_max, &global_max, 1, MPI_DOUBLE, MPI_MAX,
```

Comment:
This reduction should be done only in `rescale_cols_max`. In the `Matrix` class, columns are distributed row-wise, but rows are not distributed.